src/qat/purr/backends/live.py

```diff
@@ -213,23 +213,30 @@ def __exit__(self, exc, value, tb):
         self.shutdown()
 
     def startup(self):
+        is_connected = []
+
         for instrument in self.model.instruments.values():
-            instrument.connect()
+            is_connected.append(instrument.connect())
         if self.model.control_hardware is not None:
-            self.model.control_hardware.connect()
+            is_connected.append(self.model.control_hardware.connect())
         for baseband in self.model.basebands.values():
             if isinstance(baseband, LivePhysicalBaseband):
-                baseband.connect_to_instrument()
+                is_connected.append(baseband.connect_to_instrument())
         for physical_channel in self.model.physical_channels:
             if isinstance(physical_channel, ControlHardwareChannel):
                 for dc_bias_ch in physical_channel.dcbiaschannel_pair.values():
-                    dc_bias_ch.connect_to_instrument()
+                    is_connected.append(dc_bias_ch.connect_to_instrument())
+
+        return all(is_connected)
 
     def shutdown(self):
+        is_connected = []
```

Comment:
this is a bit confusing in terms of what true/false means here. Can we just get a quick comment to be very explicit on what true means and what you expect to return when everything is disconnected correctly.

Reply:
Great - I will see to it